src/appshell/qml/platform/linux/Main.qml

```diff
@@ -40,14 +43,112 @@ AppWindow {
         appWindow: root
     }
 
+     PLATFORM.MenuBar {
```

Comment:
I think it would be preferable to extract this into a separate component, and simply instantiate this component on both macOS and Linux. 

Also, on Linux, you could wrap it in a `Loader`, and only make this `Loader` active when the native menu bar is indeed necessary. This prevents that unnecessary components are instantiated and unnecessary property bindings and signal handlers are evaluated.